openeo/api/logs.py

```diff
@@ -0,0 +1,20 @@
+import json
+
+class LogEntry:
+    """
+    A log entry.
+    """
+
+    def __init__(self, entry):
```

Comment:
This constructor with just a generic `entry` argument API doesn't look very Pythonic or like other classes we have.
I think it is cleaner to have all fields as separate arguments to __init__, and have a classmethod `from_dict` to construct a LogEntry from a dict.

for example: https://github.com/Open-EO/openeo-python-client/blob/6b9318d87c0707cd606c0249c83dff6dcc058a02/openeo/rest/auth/oidc.py#L230-L260

Reply:
Isn't that pretty annoying to pass so many parameters individually? Also, I'm not sure how to not pass "data". There's a difference between None and not specified.

Do we actually need this wrapper? It seems pretty useless right now...

Reply:
> Isn't that pretty annoying to pass so many parameters individually?

I don't see it like that, with explicit arguments, you can easily add type hints, defaults (and documentation), which is not the case with a generic dict. You indeed have to pass the parameters individually, but the end user will not have to do that. It's probably just going to be used internally in one or two places.

> Also, I'm not sure how to not pass "data" in case it's not given. There's a difference between None (JSON: null) and not specified for data.

True, but in practice I don't think a user is going to wonder about why a field is `None`: because it was not specified by backend, or because the backend provided `null` explicitly? The practical result for the user is the same: "nothing to see here, please move along.".  The same can actually be said about `code`, `path` and `links`: now you do give them a default value if they are missing, e.g. a missing `path` will give empty list.

> Do we actually need this wrapper? It seems pretty useless right now...

The added value of a  wrapper like this are I think:
- making sure missing fields get sensible default values so the end user does not have to repeat all the related boilerplate 
- end user can access fields through attributes instead of dict keys, which is safer (because IDE can detect typos and provide code completion)
- in jupyter context you can provide "pretty" rendering

Reply:
side note: if we could drop python 3.5 support (#210 )  creating a LogEntry container would be very low on boilerplate when using [typing.NamedTuple](https://docs.python.org/3.6/library/typing.html#typing.NamedTuple)

Reply:
The README says:

> Requirement: a Python 3.6 (or higher) environment

So I'm confused.

> True, but in practice I don't think a user is going to wonder about why a field is `None`: because it was not specified by backend, or because the backend provided `null` explicitly? The practical result for the user is the same: "nothing to see here, please move along.". 

This is not correct. null (None) can actually have a specific meaning (usually `no-data`) here! This field is used for debugging and users can pass intermediate results here, in which one of the allowed values is `null` for no-data. This can be different from "doesn't exist" or "was not given/provided" by the process.

Reply:
For now: I've added a lot more documentation about each individual property and added the metrics property, too.